applications/asset_tracker_v2/src/cloud/nrf_cloud_integration.c

```diff
@@ -113,6 +113,26 @@ static void nrf_cloud_event_handler(const struct nrf_cloud_evt *evt)
 	case NRF_CLOUD_EVT_USER_ASSOCIATION_REQUEST:
 		LOG_WRN("NRF_CLOUD_EVT_USER_ASSOCIATION_REQUEST");
 		LOG_WRN("Add the device to nRF Cloud and wait for it to reconnect");
+
+		/* It is expected that the application will disconnect and reconnect to nRF Cloud
+		 * several times during device association.
+		 */
+
+		/* Explicitly disconnect the nRF Cloud transport library to clear its
```

Comment:
this comment makes it sound like the disconnect is done here only to satisfy the nrf_cloud library.
the disconnect is performed here due to how the cloud module operates (by checking the connection state after the back off timer expires).  also, because the device enters PSM and might not receive the NRF_CLOUD_EVT_USER_ASSOCIATED event. 

a disconnect/reconnect is **required by the cloud** _after_ association:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/networking/nrf_cloud.html#connecting

i don't think this comment adequately explains situation.

Reply:
Alright. The PR just got merged, but I'll make sure to add a follow-up PR fixing the comment. How about something like this:
```
/* A disconnect/reconnect is required by nRF Cloud after the
 * NRF_CLOUD_EVT_USER_ASSOCIATED event is received. This event is sent from
 * nRF Cloud when the device has been added to an nRF CLoud account.
 * However, it is likely that the device is in PSM when this occurs and not able to
 * receive this event.
 *
 * Due to this, we explicitly disconnect the application
 * when the NRF_CLOUD_EVT_USER_ASSOCIATION_REQUEST event is received and depend on
 * the reconnection routine in the cloud module to reconnect the application until
 * the device has been registered to the nRF Cloud account.
 *
 * It is expected that the application will disconnect and reconnect to nRF Cloud
 * several times during device association.
 */
```

Reply:
perfect 👍